File: LeetCode/python/1052.py
class Solution(object):
    def maxSatisfied(self, customers, grumpy, X):
        """
        :type customers: List[int]
        :type grumpy: List[int]
        :type X: int
        :rtype: int
        """
        res, n = 0, len(customers)
        for i in range(n):
            res += customers[i] * (grumpy[i] ^ 1)
        for i in range(X):
            if i >= n:
                return res
            res += customers[i] * grumpy[i]
        temp = res
        for i in range(X, n):
            temp = temp - customers[i - X] * grumpy[i - X] + customers[i] * grumpy[i]
            res = max(res, temp)
        return res

File: LeetCode/cpp/1052.cpp
class Solution {
public:
    int maxSatisfied(vector<int> &customers, vector<int> &grumpy, int X) {
        int n = customers.size(), res = 0;
        for (int i = 0; i < n; ++i)
            res += customers[i] * (grumpy[i] ^ 1);
        for (int i = 0; i < n && i < X; ++i)
            res += customers[i] * grumpy[i];
        if (X >= n)
            return res;
        int temp = res;
        for (int i = X; i < n; ++i) {
            temp = temp - customers[i - X] * grumpy[i - X] + customers[i] * grumpy[i];
            res = max(res, temp);
        }
        return res;
    }
};
